components/authz-service/server/v2/policy.go

```diff
@@ -154,18 +154,24 @@ func (s *policyServer) CreatePolicy(
 			"error parsing policy %q: %s", req.Id, err.Error())
 	}
 
-	returnPol, err := s.store.CreatePolicy(ctx, &pol)
+	returnPol, err := s.store.CreatePolicy(ctx, &pol, s.isBeta2p1())
 	switch err {
 	case nil: // continue
 	case storage_errors.ErrConflict:
 		return nil, status.Errorf(codes.AlreadyExists,
 			"policy with id %q already exists", req.Id)
 	default:
-		if err, ok := err.(*storage_errors.ForeignKeyError); ok {
+		switch err.(type) {
+		case *projectassignment.ProjectsMissingErr:
+			return nil, status.Errorf(codes.NotFound, err.Error())
```

Comment:
There's no format string here, so let's do this:
```suggestion
			return nil, status.Error(codes.NotFound, err.Error())
```